spec/setup/helpers/perf_helper_spec.rb

```diff
@@ -20,17 +21,50 @@ def options
     @options = {}
   end
 
+
 end
 
 describe PerfHelperClass do
   let!(:hosts) {[{'platform' => Beaker::Platform.new('centos-6.5-x86_64')}]}
 
+  context '.perf_init' do
+    let!(:options) { {} }
+
+    it 'it sets @install_opts to options with :dev_builds_repos option => ["PC1"]' do
+      # TODO: test that @install_opts is set?
+      allow(subject).to receive(:options).and_return(options)
+      expect(subject.options).to receive(:merge).with({:dev_builds_repos => ["PC1"]})
+      subject.perf_init
+    end
+
+  end
+
+  context '.set_etc_hosts' do
+    let!(:master) {[]}
+
+    it 'it sets the puppet ip in /etc/hosts' do
+
+      # TODO: is this the best way to handle loadbalancer.ip / master.ip
+      master_ip = '127.0.0.1'
+      allow(subject).to receive(:hosts).and_return(hosts)
+      allow(master).to receive(:ip).and_return(master_ip)
+      allow(subject).to receive(:master).and_return(master)
+
+      # TODO: test both true and false (loadbalancer.ip / master.ip)?
+      expect(subject).to receive(:any_hosts_as?).with((:loadbalancer)).and_return(false)
+      expect(subject).to receive(:on).with(hosts, "echo \"#{master_ip} puppet\" >> /etc/hosts").once
+      subject.set_etc_hosts
+
+    end
+
+  end
+
```

Comment:
You're going to hate me. In my check-in I'm about to push up for a PR, I removed both of these functions. I found a better way to handle the perf_init, and the set_etc_hosts is no longer necessary with the new AWS account.
With that said, I think the approach is perfect, If we had similar tests for all of the other functions we'd be in great shape.